### DISCUSSION TITLE
How to get text of the child element?

### DISCUSSION THREAD

**Discussion by Harsha-Gelato** [appium-userland/appium-flutter-driver#574] [General]
Hi Team,
I am using appium flutter driver, can someone please tell me how can I get text from the child element?

**My current code is:**
     await driver.execute('flutter:waitFor', find.byValueKey(`line_graph`)); 

        btn = find.descendant({
            of: find.byValueKey('line_graph'),
            matching: find.byValueKey('test'),
            matchRoot: false,
            firstMatchOnly: true,
        });
        await driver.execute('flutter:waitFor', btn);
        const firstdate: string = await driver.getElementText(btn);
    
console.log(firstdate);

**Error:**
<img width="999" alt="Screenshot 2023-09-20 at 11 37 00" src="https://github.com/appium/appium-flutter-driver/assets/103107117/493811ed-b713-4141-9b5f-7370f22b6719">

**strong text**Code of key:
<img width="648" alt="Screenshot 2023-09-20 at 11 37 27" src="https://github.com/appium/appium-flutter-driver/assets/103107117/a1ac9b71-9747-49e9-8119-497985219b65">
<img width="375" alt="Screenshot 2023-09-20 at 09 38 50" src="https://github.com/appium/appium-flutter-driver/assets/103107117/92453d30-0c8f-4ff9-8aaf-5cdde37a4a1f">

**Comment by Harsha-Gelato** [comment]
@KazuCocoa Could you please share some thoughts on this? Thanks :)

**Reply by KazuCocoa** [reply]
I guess the Flutter's error message indicated the element did not support getText by flutter_driver. Could it see the element/text with `flutter: getRenderTree` command?

**Reply by Harsha-Gelato** [reply]
@KazuCocoa Could you please tell me what is the best way I can print full output of getRenderTree command in some file because the output is too long and when I run this command in my script, half of the output gets skipped due to which I cannot share you the full output?

**Reply by KazuCocoa** [reply]
I think you can just write the entire string response in a file like https://www.geeksforgeeks.org/writing-to-file-in-python/ in the client side

**Reply by Harsha-Gelato** [reply]
[test.txt](https://github.com/appium/appium-flutter-driver/files/12748282/test.txt)
@KazuCocoa Please find the output :)

**Reply by KazuCocoa** [reply]
`line_graph` had nested element with `line_graph` key. I'm not so familiar with the app structure, but I wonder if it needs to tune key somehow to reduce nest between `line_graph` and `test`, for example...
https://gist.github.com/KazuCocoa/630a6554bc6ee12c7f0de49478405a51

**Reply by Harsha-Gelato** [reply]
okay thankyou for your input. I will see how can I reduce this nesting :)
